src/main/fc/fc_tasks.c

```diff
@@ -114,29 +115,44 @@ static void taskHandleSerial(timeUs_t currentTimeUs)
     mspSerialProcess(ARMING_FLAG(ARMED) ? MSP_SKIP_NON_MSP_DATA : MSP_EVALUATE_NON_MSP_DATA, mspFcProcessCommand);
 }
 
-static void taskUpdateBattery(timeUs_t currentTimeUs)
+
+void taskBatterySensors(timeUs_t currentTimeUs)
 {
-#if defined(USE_ADC) || defined(USE_ESC_SENSOR)
-    if (feature(FEATURE_VBAT) || feature(FEATURE_ESC_SENSOR)) {
-        static uint32_t vbatLastServiced = 0;
+    static uint32_t vbatLastServiced = 0;
+
+    if (batteryConfig()->voltageMeterSource != VOLTAGE_METER_NONE) {
         if (cmp32(currentTimeUs, vbatLastServiced) >= VBATINTERVAL) {
             vbatLastServiced = currentTimeUs;
-            updateBattery();
+
+            batteryUpdateVoltage();
         }
     }
-#endif
 
-    if (feature(FEATURE_CURRENT_METER) || feature(FEATURE_ESC_SENSOR)) {
+    if (batteryConfig()->currentMeterSource != CURRENT_METER_NONE) {
```

Comment:
Instead of checking the `VBATINTERVAL` and the `IBATINTERVAL`, why not just have a task for IBAT and one for VBAT and schedule each appropriately?